src/rust/engine/src/externs/mod.rs

```diff
@@ -212,13 +212,18 @@ pub fn getattr_from_str_frozendict<'p, T: FromPyObject<'p>>(
 }
 
 pub fn getattr_as_optional_string(value: &PyAny, field: &str) -> Option<String> {
-  let v = value.getattr(field).unwrap();
-  if v.is_none() {
-    return None;
+  // let v = value.getattr(field).unwrap();
+  if let Ok(v) = value.getattr(field) {
+    if v.is_none() {
+      return None;
+    }
+    if let Ok(inner) = v.extract() {
+      return Some(inner);
+    }
   }
   // TODO: It's possible to view a python string as a `Cow<str>`, so we could avoid actually
   // cloning in some cases.
-  Some(v.extract().unwrap())
+  None
```

Comment:
Rather than returning `None`, it would be good to actually change the return type of this function to: `PyResult<Option<String>>`, so that if the initial `getattr` fails, it can be returned.